### PR TITLE
Add edit and remove functionality to Detail View.

### PR DESCRIPTION
**Pull request by thealmightyone**
ESI-9821

### CODE REVIEW

File: src/DetailView/DetailActions.js [hunk 1]
```diff
@@ -127,3 +140,104 @@ export function cancelPollData() {
...
+        break;
+      case 404:
+        error = 'Resource has not been found.';
+        break;
+      case 409:
+        error = 'Conflict.';
+    }
+  } else {
+    error = error.toString();
+  }
```
Comment (sealionkat) [inline src/DetailView/DetailActions.js]: There is some functionality repeated in updateError (maintaining error).

File: src/components/Detail.jsx [hunk 2]
```diff
@@ -1,12 +1,35 @@
 import React, {Component, PropTypes} from 'react';
+import {Tooltip, Position} from '@blueprintjs/core';
 
 export default class Detail extends Component {
+  handleEditClick = () => {
+    this.props.onEdit(this.props.data);
+  };
+
+  handleRemoveClick = () => {
+    console.log(this.props.data);
```
Comment (sealionkat) [inline src/components/Detail.jsx]: Please remove this `console.log`

File: src/DetailView/DetailActions.js (hunk 1 above)
Comment (thealmightyone) [inline src/DetailView/DetailActions.js]: It looks the same now, but this is where custom error messages are handled for each action type.

File: src/components/Detail.jsx [hunk 3]
```diff
@@ -1,12 +1,34 @@
...
+  };
+
+
   render() {
     const {schema} = this.props.schema;
     const {data} = this.props;
-
     return (
       <div className="pt-card pt-elevation-3 detail">
+        <h2 style={{display: 'inline-block'}}>{schema.title}schema title</h2>
```
Comment (sealionkat) [inline src/components/Detail.jsx]: inline styles ;)

File: src/components/Detail.jsx [hunk 4]
```diff
@@ -1,12 +1,34 @@
...
+        <div className='icons'>
+          <Tooltip content='Edit' hoverOpenDelay={50}
+            position={Position.BOTTOM}>
+            <span className="pt-icon-standard pt-icon-edit" onClick={this.handleEditClick}/>
+          </Tooltip>
+          <Tooltip content='Delete' hoverOpenDelay={50}
+            position={Position.BOTTOM}>
+            <span className="pt-icon-standard pt-icon-trash" onClick={this.handleRemoveClick}
+              style={{marginLeft: '15px'}}
```
Comment (sealionkat) [inline src/components/Detail.jsx]: inline styles

File: src/DetailView/DetailActions.js [hunk 5]
```diff
@@ -127,3 +140,104 @@ export function cancelPollData() {
...
+  };
+}
+
+function deleteSuccess() {
+  browserHistory.goBack();
+  return dispatch => {
+    dispatch({type: DELETE_SUCCESS});
+  };
+}
+function deleteError(error) {
```
Comment (stepek) [inline src/DetailView/DetailActions.js]: Please add new line before function

File: src/DetailView/DetailView.jsx [hunk 6]
```diff
@@ -2,32 +2,50 @@ import React, {Component, PropTypes} from 'react';
...
+  clearData,
+  pollData,
+  cancelPollData,
+  deleteData,
+  updateData
+} from './DetailActions';
+import Dialog from '../Dialog/Dialog';
 import axios from 'axios';
 
+import {Alert, Intent} from '@blueprintjs/core';
```
Comment (stepek) [inline src/DetailView/DetailView.jsx]: Please move up all imports in this file

File: src/DetailView/DetailActions.js [hunk 7]
```diff
@@ -127,3 +140,104 @@ export function cancelPollData() {
...
+        dispatch(updateError('Cannot update resource!'));
+      }
+    }).catch(error => {
+      dispatch(updateError(error));
+    });
+  };
+}
+
+function deleteSuccess() {
+  browserHistory.goBack();
```
Comment (thealmightyone) [inline src/DetailView/DetailActions.js]: This looks like a good enough place for now. If the can find a better place in which this like still executes before 'fetch' then we'll be able to change it.

**Comment by sealionkat** [comment]
Resolve conflicts pls.

**Pull request merged**